protocol-units/bridge/indexer-db/src/client.rs

```diff
@@ -0,0 +1,88 @@
+use crate::models::*;
+use crate::schema::*;
+use bridge_util::chains::bridge_contracts::BridgeContractEvent;
+use bridge_util::TransferActionType;
+use diesel::pg::PgConnection;
+use diesel::prelude::*;
+
+pub struct Client {
+	conn: PgConnection,
+}
+
+impl Client {
+	/// Creates a new client with the given connection.
+	pub fn new(conn: PgConnection) -> Self {
+		Self { conn }
+	}
+
+	/// Inserts a new transfer action into the database.
+	pub fn insert_transfer_action(
+		&self,
+		action_type: TransferActionType,
+	) -> Result<(), diesel::result::Error> {
+		match action_type {
+			TransferActionType::LockBridgeTransfer(_) => {
+				diesel::insert_into(transfer_actions::table)
+					.values(NewTransferAction { action_type })
+					.execute(&self.conn)?;
+			}
+			TransferActionType::WaitAndCompleteInitiator(u64, HashLockPreImage) => {
+				diesel::insert_into(transfer_actions::table)
+					.values(NewTransferAction { action_type })
+					.execute(&self.conn)?;
+			}
+			TransferActionType::RefundInitiator => {
+				// do nothing
+			}
```

Comment:
While we are here. I think there is value to storing Refund Actions in the DB also. Purely for internal reports.

Reply:
They don't have any data though. What do you want to store?

Reply:
Yeah, that's yet to be completed. We update later